crucible-mir/src/Mir/TransTy.hs

```diff
@@ -314,6 +315,26 @@ isZeroSized col ty = go ty
       _ -> False
 
 
+-- | Get the "ABI-level" function arguments for `sig`, which determines the
+-- arguments we use for the Crucible `FnHandle`.  This applies a special
+-- adjustment for `extern "rust-call"` functions, like the one rustc applies
+-- when converting a `FnSig` to `FnAbi`.
+abiFnArgs :: HasCallStack => M.FnSig -> [M.Ty]
+abiFnArgs sig = case sig ^. M.fsabi of
+  M.RustCall M.RcNoBody -> untupledArgs
+  M.RustCall M.RcNoSpreadArg -> normalArgs
+  M.RustCall (M.RcSpreadArg _) -> untupledArgs
+  _ -> normalArgs
+  where
+    normalArgs = sig ^. M.fsarg_tys
+    untupledArgs = case normalArgs of
+      [M.TyTuple tys] -> tys
+      [selfTy, M.TyTuple tys] -> selfTy : tys
+      _ -> error $
+          "unexpected argument list for " ++ show (sig ^. M.fsabi) ++  ": "
+            ++ show (sig ^. M.fsarg_tys)
```

Comment:
Consider using `show (pretty ...)` instead of `show ...` here to make the results a bit more compact and human-readable.